src/journal.c

```diff
@@ -116,119 +115,119 @@ static const UnkStruct_020E5B50 Unk_020E5B50[] = {
 
-    v2 = DayNumberForDate(&v1) - DayNumberForDate(&v0);
+    daysDiff = DayNumberForDate(&currDate) - DayNumberForDate(&journalEntryTitle);
 
-    if (((v1.month == 12) && (v1.day == 31) && (v0.month == 1) && (v0.day == 1)) || ((v1.month == 1) && (v1.day == 1) && (v0.month == 12) && (v0.day == 31))) {
-        s32 v3 = (s32)v1.year - (s32)v0.year;
+    if (((currDate.month == 12) && (currDate.day == 31) && (journalEntryTitle.month == 1) && (journalEntryTitle.day == 1)) || ((currDate.month == 1) && (currDate.day == 1) && (journalEntryTitle.month == 12) && (journalEntryTitle.day == 31))) {
+        s32 yearsDiff = (s32)currDate.year - (s32)journalEntryTitle.year;
 
-        if ((v3 >= 2) || (v3 <= -2)) {
-            return 1;
+        if ((yearsDiff >= 2) || (yearsDiff <= -2)) {
+            return TRUE;
         }
-    } else if ((v2 <= -2) || (v2 >= 2)) {
-        return 1;
+    } else if ((daysDiff <= -2) || (daysDiff >= 2)) {
+        return TRUE;
     } else {
-        if (v1.year != v0.year) {
-            return 1;
+        if (currDate.year != journalEntryTitle.year) {
+            return TRUE;
         }
     }
 
-    return 0;
+    return FALSE;
 }
 
-void Journal_SaveData(Journal *journal, void *param1, u8 param2)
+void JournalEntry_SaveData(JournalEntry *journalEntry, void *data, u8 dataType)
 {
-    if (journal != NULL) {
-        switch (param2) {
-        case 0:
-            sub_0202B7A0(journal, param1);
+    if (journalEntry != NULL) {
+        switch (dataType) {
+        case JOURNAL_TITLE:
+            JournalEntry_SaveTitle(journalEntry, data);
             break;
-        case 1:
-            sub_0202B7E0(journal, param1);
+        case JOURNAL_UNK_04:
+            sub_0202B7E0(journalEntry, data);
             break;
-        case 2:
-            sub_0202B880(journal, param1);
+        case JOURNAL_MON:
+            JournalEntry_SaveMon(journalEntry, data);
             break;
-        case 3:
-            sub_0202B88C(journal, param1);
+        case JOURNAL_TRAINER:
+            JournalEntry_SaveTrainer(journalEntry, data);
             break;
-        case 4:
-            sub_0202B898(journal, param1);
+        case JOURNAL_UNK_1C:
+            sub_0202B898(journalEntry, data);
             break;
         }
     }
 
-    Heap_FreeToHeap(param1);
+    Heap_FreeToHeap(data);
 }
 
-static void sub_0202B7A0(Journal *journal, UnkStruct_0202BC58 *param1)
+static void JournalEntry_SaveTitle(JournalEntry *journalEntry, JournalEntryTitle *journalEntryTitle)
 {
-    if ((journal->unk_00.unk_00_0 == param1->unk_00_0) && (journal->unk_00.unk_00_7 == param1->unk_00_7) && (journal->unk_00.unk_00_14 == param1->unk_00_14) && (journal->unk_00.unk_00_11 == param1->unk_00_11)) {
+    if ((journalEntry->title.year == journalEntryTitle->year) && (journalEntry->title.month == journalEntryTitle->month) && (journalEntry->title.day == journalEntryTitle->day) && (journalEntry->title.week == journalEntryTitle->week)) {
```

Comment:
**polish**: Clean up unnecessary parens and split conditionals up onto separate lines for legibility.